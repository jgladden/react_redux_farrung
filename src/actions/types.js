export const SET_SECTION = 'SET_SECTION';

export const GET_PORTFOLIO_ITEMS = 'GET_PORTFOLIO_ITEMS';
export const GET_PORTFOLIO_ITEMS_SUCCESS = 'GET_PORTFOLIO_ITEMS_SUCCESS';
export const GET_PORTFOLIO_ITEMS_ERROR = 'GET_PORTFOLIO_ITEMS_ERROR';

export const EDIT_PORTFOLIO_ITEM = 'EDIT_PORTFOLIO_ITEM';
export const EDIT_PORTFOLIO_ITEM_SUCCESS = 'EDIT_PORTFOLIO_ITEM_SUCCESS';
export const EDIT_PORTFOLIO_ITEM_ERROR = 'EDIT_PORTFOLIO_ITEM_ERROR';

export const ADD_PORTFOLIO_ITEM = 'ADD_PORTFOLIO_ITEM';
export const ADD_PORTFOLIO_ITEM_SUCCESS = 'ADD_PORTFOLIO_ITEM_SUCCESS';
export const ADD_PORTFOLIO_ITEM_ERROR = 'ADD_PORTFOLIO_ITEM_ERROR';

export const REMOVE_PORTFOLIO_ITEM = 'REMOVE_PORTFOLIO_ITEM';
export const REMOVE_PORTFOLIO_ITEM_SUCCESS = 'REMOVE_PORTFOLIO_ITEM_SUCCESS';
export const REMOVE_PORTFOLIO_ITEM_ERROR = 'REMOVE_PORTFOLIO_ITEM_ERROR';

export const POST_AUTH = 'POST_AUTH';
export const POST_AUTH_SUCCESS = 'POST_AUTH_SUCCESS';
export const POST_AUTH_ERROR = 'POST_AUTH_ERROR';
export const AUTH_LOGOUT = 'AUTH_LOGOUT';

export const POST_MESSAGE = 'POST_MESSAGE';
export const POST_MESSAGE_SUCCESS = 'POST_MESSAGE_SUCCESS';
export const POST_MESSAGE_ERROR = 'POST_MESSAGE_ERROR';
