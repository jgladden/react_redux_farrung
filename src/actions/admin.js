import axios from 'axios';
import * as types from './types';
import { getAdminUrl } from '../config';

export function fetchAdmin(token) {
  const config = {
    headers: {
      'Authorization': `bearer ${token}`
    }
  };
  return dispatch => {
    dispatch(getAdminItems());
    return axios.get(getAdminUrl, config)
      .then(response => dispatch(
        getAdminItemsSuccess(response.data))
      )
      .catch(error => dispatch(
        getAdminItemsError(error.toString()))
      );
  };
}

export const getAdminItems = () => ({
  type: types.GET_ADMIN_ITEMS
});

export const getAdminItemsSuccess = payload => ({
  type: types.GET_ADMIN_ITEMS_SUCCESS,
  payload
});

export const getAdminItemsError = payload => ({
  type: types.GET_ADMIN_ITEMS_ERROR,
  payload
});

export const setDisplayType = payload => ({
  type: types.SET_DISPLAY_TYPE,
  payload
});

export const setDisplayArchived = payload => ({
  type: types.SET_DISPLAY_ARCHIVED,
  payload
});

export const mergeAdminItem = payload => ({
  type: types.MERGE_ADMIN_ITEM,
  payload
});

export const removeAdminItem = payload => ({
  type: types.REMOVE_ADMIN_ITEM,
  payload
});
