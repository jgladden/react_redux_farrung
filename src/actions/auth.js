import axios from 'axios';
import * as types from './types';
import { postAuthUrl } from '../config';

export function submitLogin(authObj) {
  return dispatch => {
    dispatch(postAuth());
    return axios.post(postAuthUrl, authObj)
      .then(response => {
        let error = response.data.error;
        if(!error)
          return dispatch(
            postAuthSuccess(response.data.token)
          );
        return dispatch(
          postAuthError(error)
        );
      })
      .catch(error => dispatch(
        postAuthError(error.toString()))
      );
  };
}

export const postAuth = () => ({
  type: types.POST_AUTH
});

export const postAuthSuccess = payload => ({
  type: types.POST_AUTH_SUCCESS,
  payload
});

export const postAuthError = payload => ({
  type: types.POST_AUTH_ERROR,
  payload
});

export const submitLogout = () => ({
  type: types.AUTH_LOGOUT
});

