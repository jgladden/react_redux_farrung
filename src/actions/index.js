import { jsonFetch } from '../utils/jsonFetch';
import { 
  ADD_PORTFOLIO_TYPE,
  SET_PORTFOLIO_TYPE,
  ADD_PORTFOLIO_ITEM, 
  REMOVE_PORTFOLIO_ITEM 
} from '../constants/action_types';

const api = 'http://farrung.com/api/portfolio1/';

export function fetchPortfolioType(type) {
  let url = api + type;
  return function(dispatch) {
    return jsonFetch(url).then(data => {
      dispatch(addPortfolioType(data));
    });
  };
}

export const addPortfolioType = payload => ({
  type: ADD_PORTFOLIO_TYPE,
  payload
});


export const setPortfolioType = payload => ({
  type: SET_PORTFOLIO_TYPE,
  payload
});

export const addPortfolioItem = payload => ({
  type: ADD_PORTFOLIO_ITEM,
  payload
});

export const removeTodo = payload => ({
  type: REMOVE_PORTFOLIO_ITEM,
  payload
});


