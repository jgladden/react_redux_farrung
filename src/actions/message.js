import axios from 'axios';
import * as types from './types';
import { postMessageUrl } from '../config';

export function submitMessage(messageObj) {
  return dispatch => {
    dispatch(postMessage());
    return axios.post(postMessageUrl, messageObj)
      .then(response => dispatch(
        postMessageSuccess(response.data))
      )
      .catch(error => dispatch(
        postMessageError(error.toString()))
      );
  };
}

export const postMessage = () => ({
  type: types.POST_MESSAGE
});

export const postMessageSuccess = payload => ({
  type: types.POST_MESSAGE_SUCCESS,
  payload
});

export const postMessageError = payload => ({
  type: types.POST_MESSAGE_ERROR,
  payload
});

