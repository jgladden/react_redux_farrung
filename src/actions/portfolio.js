import axios from 'axios';
import * as types from './types';
import { 
  getPortfolioUrl 
} from '../config';

export function fetchPortfolioItems() {
  return dispatch => {
    dispatch(getPortfolioItems());
    return axios.get(getPortfolioUrl)
      .then(response => dispatch(
        getPortfolioItemsSuccess(response.data))
      )
      .catch(error => dispatch(
        getPortfolioItemsError(error.toString()))
      );
  };
}

export const getPortfolioItems = () => ({
  type: types.GET_PORTFOLIO_ITEMS
});

export const getPortfolioItemsSuccess = payload => ({
  type: types.GET_PORTFOLIO_ITEMS_SUCCESS,
  payload
});

export const getPortfolioItemsError = payload => ({
  type: types.GET_PORTFOLIO_ITEMS_ERROR,
  payload
});


export const mergePortfolioItem = payload => ({
  type: types.MERGE_PORTFOLIO_ITEM,
  payload
});

export const removePortfolioItem = payload => ({
  type: types.REMOVE_PORTFOLIO_ITEM,
  payload
});

