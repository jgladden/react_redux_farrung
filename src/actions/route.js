import * as types from './types';

const processRoute = url => {
  if(!url) return null;
  let urlParts = url.replace(/^\//, '').replace(/\/$/, '').split('/');
  if(!urlParts[1]) {
    if(urlParts[0] === 'portfolio')
      urlParts[1] = 'online';
    if(urlParts[0] === 'admin')
      urlParts[1] = 'list';
  }
  url = `/${urlParts.join('/')}`;
  return {  url, urlParts };
};


export const setRoute = url => {
  let payload = processRoute(url);
  return {
    type: types.SET_ROUTE,
    payload
  };
};

