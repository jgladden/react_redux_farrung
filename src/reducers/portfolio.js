import {
  FETCH_PORTFOLIO,
  ADD_PORTFOLIO_ITEM,
  REMOVE_PORTFOLIO_ITEM
} from '../constants/action_types';

const portfolio = (state = {}, action) => {
  switch (action.type) {
  case FETCH_PORTFOLIO: {
    return ({
      ...action.payload.portfolio
    });
  }
  case ADD_PORTFOLIO_ITEM: {
    let { type, item } = action.payload;
    let portfolio = {...state};
    portfolio[type][item.id] = item;
    return {
      ...portfolio
    };
  }
  case REMOVE_PORTFOLIO_ITEM: {
    let { type, id } = action.payload;
    let portfolio = {...state};
    delete portfolio[type][id];
    return {
      ...portfolio
    };
  }
  default:
    return state;
  }
};

export default portfolio;
