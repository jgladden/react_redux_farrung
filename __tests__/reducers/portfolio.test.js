import portfolio from '../../src/reducers/portfolio';
import * as types from '../../src/actions/types';

describe('1display portfolio reducer', () => {
  test('initial state is correct', () => {
    const action = { type: '' };
    expect(portfolio(undefined, action)).toMatchSnapshot();
  });
  test('FETCH_PORTFOLIO_REQUEST returns expected state', () => {
    const action = { type: types.FETCH_PORTFOLIO_REQUEST };
    expect(portfolio(undefined, action)).toMatchSnapshot();
  });
  test('FETCH_PORTFOLIO_SUCCESS returns expected state', () => {
    const action = { type: types.FETCH_PORTFOLIO_SUCCESS, payload: { data: 1 } };
    expect(portfolio(undefined, action)).toMatchSnapshot();
  });
  test('FETCH_PORTFOLIO_FAILURE returns expected state', () => {
    const action = { type: types.FETCH_PORTFOLIO_FAILURE, payload: Error('msg') };
    expect(portfolio(undefined, action)).toMatchSnapshot();
  });
});
