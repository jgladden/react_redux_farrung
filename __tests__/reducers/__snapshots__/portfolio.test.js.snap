// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1display portfolio reducer FETCH_PORTFOLIO_FAILURE returns expected state 1`] = `
Object {
  "error": "Error: msg",
}
`;

exports[`1display portfolio reducer FETCH_PORTFOLIO_REQUEST returns expected state 1`] = `
Object {
  "fetching": 1,
}
`;

exports[`1display portfolio reducer FETCH_PORTFOLIO_SUCCESS returns expected state 1`] = `
Object {
  "items": Object {
    "data": 1,
  },
}
`;

exports[`1display portfolio reducer initial state is correct 1`] = `Object {}`;
